Cubical/Algebra/Ring/Base.agda

```diff
@@ -240,6 +241,60 @@ uaRing {A = A} {B = B} = equivFun (RingPath A B)
 isGroupoidRing : isGroupoid (Ring ℓ)
 isGroupoidRing _ _ = isOfHLevelRespectEquiv 2 (RingPath _ _) (isSetRingEquiv _ _)
 
+open RingStr
+open IsRingHom
+
+-- TODO: Induced structure results are temporarily inconvenient while we transition between algebra
+-- representations
+module _ (R : Ring ℓ) {A : Type ℓ}
+  (0a 1a : A)
+  (add mul : A → A → A)
+  (inv : A → A)
+  (e : ⟨ R ⟩ ≃ A)
+  (p0 : e .fst (R .snd .0r) ≡ 0a)
+  (p1 : e .fst (R .snd .1r) ≡ 1a)
+  (p+ : ∀ x y → e .fst (R .snd ._+_ x y) ≡ add (e .fst x) (e .fst y))
+  (p· : ∀ x y → e .fst (R .snd ._·_ x y) ≡ mul (e .fst x) (e .fst y))
+  (pinv : ∀ x → e .fst (R .snd .-_ x) ≡ inv (e .fst x))
+  where
+
+  private
+    module R = RingStr (R .snd)
+
+    BaseΣ : Type (ℓ-suc ℓ)
+    BaseΣ = Σ[ B ∈ Type ℓ ] B × B × (B → B → B) × (B → B → B) × (B → B)
+
+    FamilyΣ : BaseΣ → Type ℓ
+    FamilyΣ (B , u0 , u1 , a , m , i) = IsRing u0 u1 a m i
+
+    inducedΣ : FamilyΣ (A , 0a , 1a , add , mul , inv)
+    inducedΣ = {!!}
```

Comment:
@ecavallo I tried to define InducedRing, but Agda gets extremely slow when I try to fill this hole with the code below... Any clue why?

Reply:
What I've noticed is that there seems to be an exponential explosion when running `autoUARel BaseΣ` as  `BaseΣ`  gets bigger

Reply:
Here I am not sure if it plays but you have not finished to define `FamilyΣ` but you are already using it. 
When I do that it often gets slow. I guess that agda must be trying to infer some constraints and then take time to see that you don't need the definition for it to work. 
You can try commenting the rest, fill the hole then uncomment ?
It can sometimes speed up the code drastically

Reply:
Yeah I know, it doesn't matter if I uncomment everything else in the file. I've localized the problem to autoUARel as I wrote above. It seems like the time it takes for the automation is exponential in the size of the input signature...